src/flamegraph/color/palettes.rs

```diff
@@ -99,6 +99,25 @@ pub(super) mod wakeup {
     }
 }
 
+pub(super) mod rust {
+    use crate::flamegraph::color::BasicPalette;
+
+    // Partition into user code and Rust system code
+    pub fn resolve(name: &str) -> BasicPalette {
+        if name.starts_with("core::") || name.starts_with("std::") {
+            return BasicPalette::Orange;
+        } else if name.contains("as core::")
+            || name.contains("as std::")
+            || name.contains("as alloc::")
+        {
+            return BasicPalette::Orange;
+        } else if name.contains("::") {
+            return BasicPalette::Yellow;
```

Comment:
What case is this for? That is, what's the rationale for coloring these differently than things without `::`?

Reply:
That's a catch-all for any rust function that isn't part of `core`, `alloc`, or `std`.  So I'm assuming it's a function from the user's code being tested.  I see now that it would also match on functions from dependency crates, but generally I see it as a way for a user to locate code that they wrote in the flamegraph.

Is there a better way to identify Rust functions?  From scanning my test flamegraphs, I don't see any "::" in the lower-level functions (e.g. `cpal_alsa_in`, `start_thread`), so it's been working on my limited samples.  Doesn't seem robust, though.

Other options include using `regex` patterns, and possibly even parsing `Cargo.toml` to find crate names to match on, so dependencies could look different from project crates.

Reply:
Ohh, I read this entirely backwards then. `::` is "any Rust function", and the _else_ case is for libc/C/syscall type stuff. I totally missed that. Would be good to call that out with a comment I think.

As for whether it's right, I think it depends on things like what language you're interfacing _with_. For example, if Rust calls into C++ code (or vice-versa), a bunch of C++ methods would also be colored with the "Rust color" here. But maybe that's okay. I guess what I was getting at with my question was more "should we just make anything that _isn't_ core/alloc/std have the same color?".

Reply:
I feel like it would be useful to partition the Rust functions into user/system, mostly so it's easy for a user to find their own code in the flamegraph.  I don't have the experience to make the call, but it seems like these false positives would be okay.  I think code from different languages would appear grouped in distinct regions of the flamegraph, so it wouldn't reduce readability.

I'll have time later this evening to test more examples.  I'm currently just using a pure Rust example for testing.

Reply:
Yeah, that makes sense — I buy it.